funflow-tutorial/README.md

```diff
@@ -2,19 +2,49 @@
 
 This folder holds the tutorials for `funflow`. They are written as a series of Jupyter notebooks.
 
+<!-- toc -->
+
+- [About](#about)
+- [Setup from scratch](#setup-from-scratch)
+  * [Snags and solutions](#snags-and-solutions)
+- [Development](#development)
+
+<!-- tocstop -->
+
 ## About
 
 The tutorials are meant to progressively teach new users how to use the library and tools.
 Tutorials are published as a part of the `funflow` documentation. They can be run locally using
 the Jupyter Lab environment and IHaskell kernel defined in [shell.nix](./shell.nix):
-
+<a name="run-nbs"></a>
 ```console
 $ nix-shell
 $ jupyter lab
 ```
 
+
+## Setup from scratch
+Here are a few steps and tips to avoid snags as you get started. Note that step 3 onward applies generally, for subsequent use even after installation/setup. Essentially, this section clarifies what to do before [running a notebook](#run-nbs), and what those commands are doing.
+1. **Install Nix**, per the [Nix download page](https://nixos.org/download.html).
+2. **Install `stack`**, per the [Stack website](https://docs.haskellstack.org/en/stable/install_and_upgrade/)
```

Comment:
Is stack necessary for running the notebooks? AFAIK the build should get handled in nix.

Reply:
Oh good point, I guess I installed stack anyway, figuring I'd need it to `stack build` and `stack test` things, but I'm not sure. The first few tries starting up IHaskell I couldn't import things, like `import Funflow.Tasks.Docker (DockerTaskConfig)` would fail, but it seemed like once I `stack build` the project, then it worked. It could've been some other factors, though, like maybe I wasn't starting the notebook from Nix shell. 

More generally, is there a way through Nix to build and test it, and is that the preferred way? I've been doing things like `stack build` and `stack test`...

Reply:
Hmm that is odd since the jupyter environment should include funflow ([link](https://github.com/tweag/funflow2/blob/master/nix/pkgs/jupyter.nix#L10)). If you have a chance, could you try launching jupyter from the nix shell and seeing if that works? You can run `nix-shell --pure` if you want to really make sure that no additional stuff from the local environment is visible.

For building the project, we are using nix as the primary build tool (though I also use stack locally sometimes). To build individual components you can run `nix-build -A someattribute` from the repo root. The default.nix contains a full list of the available build attributes: https://github.com/tweag/funflow2/blob/master/default.nix.

Reply:
Thanks a bunch for the Nix pointers! Yeah, it's working fine now with `--pure`. I think the key point is which kernel is running. When I start jupyter, I see both `Haskell` and `Haskell - haskell` as options, both on the sort of "home" / landing page when Jupyter starts (i.e., an icon for each of those), and each is visible under "Start Other Kernel" dropdown when creating a new notebook. When I select plain `Haskell`, things don't work (i.e., can't import funflow things, or even do any Haskell stuff, at least with `--pure`), seemingly due to lack of GHC (as expected, I guess?) based on output to the terminal where jupyter's running. All's good when using `Haskell - haskell` as kernel, though.

Reply:
Will modify accordingly